src/plugins/footnotes/index.ts

```diff
@@ -90,6 +102,108 @@ export const uncitedFootnoteWidget = () => () => {
   element.innerHTML = alertIcon
   return element
 }
+interface inlineFootnote {
+  node: InlineFootnoteNode | null
+  pos: number | null
+}
+
+const getInlineFootnote = (view: EditorView, id: string): inlineFootnote => {
+  let inlineFootnote: InlineFootnoteNode | null = null
+  let footnotePosition: number | null = null
+
+  view.state.doc.descendants((node, pos) => {
+    const footnote = node as InlineFootnoteNode
+
+    if (footnote.attrs.rids?.includes(id)) {
+      inlineFootnote = footnote
+
+      footnotePosition = pos
+      return false
+    }
+  })
+
+  return { node: inlineFootnote, pos: footnotePosition }
+}
+
+const deleteFootnoteWidget =
+  (
+    node: ManuscriptNode,
+    props: PluginProps,
+    footnoteType: string,
+    footnoteMessage: string,
+    id: string
+  ) =>
+  (view: EditorView, getPos: () => number | undefined) => {
+    const deleteBtn = document.createElement('span')
+    deleteBtn.className = 'delete-icon'
+    deleteBtn.innerHTML = deleteIcon
+
+    deleteBtn.addEventListener('click', () => {
+      const handleDelete = () => {
+        const tr = view.state.tr
+        const pos = getPos()
+
+        if (node.type === schema.nodes.table_element_footer) {
+          let isAllGeneralFootnotes = true
+
+          node.content.forEach((item) => {
+            if (item.type !== schema.nodes.paragraph) {
+              isAllGeneralFootnotes = false
+
+              return
+            }
+          })
+          if (isAllGeneralFootnotes && pos) {
+            // All child nodes are general footnotes
+            tr.delete(pos, pos + node.nodeSize - 1)
+          } else {
+            node.content.forEach((item) => {
+              if (item.type === schema.nodes.paragraph && pos) {
+                tr.delete(pos - 1, pos + item.nodeSize + 1)
+              }
+            })
+          }
+        }
+        // delete table footnotes
+        // TO Do : delete parent node "footnotes_element"
```

Comment:
A parent `foototes-element` & `table_element_footer` should be deleted once needed. 

Why this is not covered already ?

Reply:
By default, the table_element_footer gets deleted once all its child elements are deleted. However, removing the footnotes-element isn't straightforward, and doing so at the moment might complicate things and potentially necessitate transformer changes. Since it's not currently impacting anything, I suggest we add it to the to-do list for the next phase of implementation.

Reply:
Can you explain why it's not straightforward? what issues you are facing?

Reply:
Based on my conversation with Betamony,  these nodes should be deleted by default in the processmirror. Additionally, according to my discussion with Natshe today , it's acceptable to have empty nodes as they don't impact the exported JATS. We already handle similar cases, such as keywords; when all keywords are deleted, the parent node isn't automatically deleted and remains in the editor and same for contributors and affiliations. In the current scenario, the footnote_element node is deleted when all the children of table_footer are deleted.
/home/nghazzawi/Videos/Screencasts/Screencast from 04-15-2024 06:50:39 PM.webm

Reply:
I'm ok with that but this will violate some assumptions made on the code that if there is a footnote-element then it has content and we open the modal to show the content.

see: https://github.com/Atypon-OpenSource/manuscripts-body-editor/blob/master/src/lib/context-menu.ts#L311

This will lead to opening the modal but the modal will be empty. 

To fix that, you need to check if that footnote-element has any children.